awesome/database/migrations/20230825155632_023_insert_directus_fields_for_TutorialTags_and_UserTutorials_tables.js

```diff
@@ -0,0 +1,220 @@
+const { TABLES, DIRECTUS_TABLES } = require("../constants");
+const intermediateTables = {
+    TUTORIAL_TAGS: TABLES.TUTORIAL_TAGS,
+    USER_TUTORIALS: TABLES.USER_TUTORIALS,
+};
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = async function (knex) {
+    const exists = await knex.schema.hasTable(DIRECTUS_TABLES.DIRECTUS_FIELDS);
+    if (exists) {
+        await Promise.all(
+            Object.values(intermediateTables).map(async (v) => {
+                return new Promise((resolve, reject) => {

```

Comment:
Please create an util func for this. And replace the similar code in 20230815040913_012_insert_directus_fields_constrain_for_ids. Please do this to all your code below. And Please always think about creating functions when you write repeated code.

Reply:
done